Source/WebKit/UIProcess/mac/WebViewImpl.mm

```diff
@@ -4198,7 +4198,7 @@ static bool matchesExtensionOrEquivalent(const String& filename, const String& e
 
     [types addObjectsFromArray:archiveBuffer ? PasteboardTypes::forImagesWithArchive() : PasteboardTypes::forImages()];
     [pasteboard declareTypes:types.get() owner:m_view.getAutoreleased()];
-    setFileAndURLTypes(filename, extension, title, url, visibleURL, pasteboard);
+    setFileAndURLTypes(filename, extension, image->uti(), title, url, visibleURL, pasteboard);
```

Comment:
is `image` guaranteed to be non-null here?

Reply:
I'm pretty sure it is, since the only call site:

```
void PageClientImpl::setPromisedDataForImage(const String& pasteboardName, Ref<FragmentedSharedBuffer>&& imageBuffer, const String& filename, const String& extension, const String& title, const String& url, const String& visibleURL, RefPtr<FragmentedSharedBuffer>&& archiveBuffer, const String& originIdentifier)
{
    auto image = BitmapImage::create();
    image->setData(WTFMove(imageBuffer), true);
    m_impl->setPromisedDataForImage(image.ptr(), filename, extension, title, url, visibleURL, archiveBuffer.get(), pasteboardName, originIdentifier);
}
```

...passes in an `Image*` from a `Ref<BitmapImage>`. A couple of observations on that note:

- The `Image*` should probably be changed to an `Image&` to reflect this.
- There's code above that already grabs the `Image::uti()`; maybe we can store that into a local variable and use it in both places.

Reply:
Yes, `image` is guaranteed to be non-null here as @whsieh posits. I've also refactored a little bit to:

- Pass in an `WebCore::Image&` to `WebViewImpl::setPromisedDataForImage`
- Store `WebCore::Image::uti()` in a local variable.